### ISSUE TITLE
Supporting Q2 Q4 mixed model download & how to evaluate perf between it & Q2

### ISSUE THREAD

**Issue by Flux159** [antirez/ds4#226]
Are there any benchmarks of the mixed Q2 Q4 model comparing to the Q2 model? [DeepSeek-V4-Flash-Layers37-42Q4KExperts-OtherExpertLayersIQ2XXSGateUp-Q2KDown-AProjQ8-SExpQ8-OutQ8-chat-v2-imatrix-fixed.gguf](https://huggingface.co/antirez/deepseek-v4-gguf/blob/main/DeepSeek-V4-Flash-Layers37-42Q4KExperts-OtherExpertLayersIQ2XXSGateUp-Q2KDown-AProjQ8-SExpQ8-OutQ8-chat-v2-imatrix-fixed.gguf). It was mentioned in this [tweet](https://x.com/antirez/status/2057140498632950240?s=20) about generating an interpreted language. Should it be supported via the download script for 128GB machines?

Ideally ds4-eval or some other eval harness would give some information on if the Q2 Q4 model is actually better for real world coding examples?

Documenting for people who want to try out Q2 Q4:
```shell
hf download antirez/deepseek-v4-gguf DeepSeek-V4-Flash-Layers37-42Q4KExperts-OtherExpertLayersIQ2XXSGateUp-Q2KDown-AProjQ8-SExpQ8-OutQ8-chat-v2-imatrix-fixed.gguf --local-dir ./gguf
ln -sf ./gguf/DeepSeek-V4-Flash-Layers37-42Q4KExperts-OtherExpertLayersIQ2XXSGateUp-Q2KDown-AProjQ8-SExpQ8-OutQ8-chat-v2-imatrix-fixed.gguf ds4flash.gguf
```

If you already downloaded Q2 via the download script, then to go back to Q2:
```
ln -sf ./gguf/DeepSeek-V4-Flash-IQ2XXS-w2Q2K-AProjQ8-SExpQ8-OutQ8-chat-v2-imatrix.gguf ds4flash.gguf
```

**Comment by Flux159** [comment]
Downloading Q2-Q4 is documented as of https://github.com/antirez/ds4/commit/747cc949b877d27cc91976cd3c2eab68d85446c4 and it looks like #255 is adding a speed benchmark on M5 Maxes for the Q2-Q4 version. These pretty much close most of this issue once they land.

Specifically, this works now if you pull from main:
```
./download_model.sh q2-q4-imatrix 
```

There doesn't seem to be a current eval in the repo for correctness comparisons / agentic evaluation at the moment to compare models, but assuming that that will be needed at some point in the future to determine which models to continue supporting.

**Comment by antirez** [comment]
Hi, there is no need to benchmark Q2 vs Q4 vs Q2-Q4. Q4 follows token-exact more continuations from the official API, and Q2-Q4 has less logits drift compared to Q2. So for sure it's better if you have the memory. It's a tradeoff. Benchmarking this subtle difference is almost impossible, even if we have ds4-eval. You would need to repeat the benchmark at non zero temperature, and is still a narrow test.
